daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/ElementBase.scala

```diff
@@ -1026,13 +1034,15 @@ trait ElementBase
             )
             (r.minLengthValue, r.maxLengthValue)
           }
-          case (_, _, LengthKind.Implicit) =>
+          case (false, _, _, LengthKind.Implicit) =>
             SDE(
               "When lengthKind='implicit', both minLength and maxLength facets must be specified.",
             )
-          case (false, true, _) => (zeroBD, r.maxLengthValue)
-          case (false, false, _) => (zeroBD, unbBD)
-          case (true, false, _) => (r.minLengthValue, unbBD)
+          case (false, false, true, _) => (zeroBD, r.maxLengthValue)
+          case (false, false, false, _) => (zeroBD, unbBD)
+          case (false, true, false, _) => (r.minLengthValue, unbBD)
+          case (true, _, _, _) =>
+            SDE("Facet length cannot be defined with minLength and maxLength facets")
```

Comment:
This is a minor organizational thing, but it might be helpful to move this case right after the `(true, false,false, _)`case. That way the first two cases are when `hasLength` is true--we either use it or error if either minLength/maxLength is defined. That handles all the cases where hasLength is true, so it's maybe a little more mentally easier to think about the more complex minLength/maxLength cases that follow, since we don't have to consider hasLength anymore.

Also, should this be an assert.invariantFailed, since xerces should fail before we get here if length is defined with a minLength/maxLength?